src/charm.py

```diff
@@ -199,6 +237,33 @@ def configure_workload_container(self, event) -> None:
                 logger.info(msg)
                 self.unit.status = MaintenanceStatus(msg)
                 container.start(CONTAINER_NAME)
+
+            exporter_container = self.unit.get_container(EXPORTER_CONTAINER_NAME)
+            if exporter_container.can_connect():
+                services = exporter_container.get_plan().to_dict().get("services", {})
+                if services != exporter_config["services"]:
+
+                    msg = "Updating exporter pebble layer config"
+                    logger.info(msg)
+                    self.unit.status = MaintenanceStatus(msg)
+                    exporter_container.add_layer(
+                        EXPORTER_CONTAINER_NAME, exporter_config, combine=True
+                    )
+
+                    service = exporter_container.get_service("nginx-exporter")
+                    if service.is_running():
```

Comment:
Why stopping the exporter?

Reply:
tried to imitate what I was doing for the other container, but will abandon this in favor of replan_services()